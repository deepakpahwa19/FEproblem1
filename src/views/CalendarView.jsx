import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

export const CalendarView = ({ month, monthName }) => {
    const [today, setTodayDate] = useState(new Date());

    useEffect(() => {
        setInterval(() => {
            if (today !== new Date()) setTodayDate(new Date());
        }, 1000);
    }, []);

    return (
        <CalendarStyle>
            <Header>
                <h4>April 2021</h4>
                <div>icons</div>
            </Header>
            <div>
                <Table>
                    <Thead>
                        <TR>
                            <TD>S</TD>
                            <TD>M</TD>
                            <TD>T</TD>
                            <TD>W</TD>
                            <TD>T</TD>
                            <TD>F</TD>
                            <TD>S</TD>
                        </TR>
                    </Thead>
                    <tbody>
                        {month.map(week => (
                            <TR>
                                {week.map(date => (
                                    <TD currentDay={today.getDate() === date}>{date}</TD> //
                                ))}
                            </TR>
                        ))}
                    </tbody>
                </Table>
            </div>
        </CalendarStyle>
    );
};

const CalendarStyle = styled.div`
    width: 400px;
    height: 400px;
    background-color: #e9ece2;
    border: 1px solid grey;
    margin: 10px auto;
    padding: 20px;
`;

const Header = styled.div`
    height: 30px;
    margin-bottom: 20px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
`;

const Table = styled.table`
    width: 100%;
`;

const Thead = styled.thead``;

const TR = styled.tr`
    width: 100%;
    display: flex;
    margin: 10px 0 20px;
    justify-content: space-between;
`;

const TD = styled.td`
    width: 55px;
    background-color=${props => (props.currentDay ? 'red' : '')}
`;
