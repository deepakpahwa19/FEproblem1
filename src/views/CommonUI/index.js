export { H4, H1, H3 } from './HeaderGroup';
export { RequiredElement } from './RequiredElement';
export { Button } from './Button';
