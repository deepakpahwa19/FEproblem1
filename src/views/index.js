export { DropDownView } from './DropdownView/DropDownView';
export { RadioButtonView } from './RadioButtonView/RadioButtonView';
export { FlexContainer } from './StyledComponents/FlexContainer';
export { JourneyCardView } from './JourneyCardView/JourneyCardView';
export { H1, H4, H3, RequiredElement } from './CommonUI';
export { FooterView } from './Footer/Footer';
export { Spinner } from './SpinnerView/Spinner';
export { Button } from './CommonUI/ButtonView';
export { ErrorMessage } from './CommonUI/ErrorMessage';
export { Backdrop } from './Backdrop/Backdrop';
