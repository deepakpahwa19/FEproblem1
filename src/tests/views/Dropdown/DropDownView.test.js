import { mount } from 'enzyme';
import { DropDownView } from '../../../views';

describe('DropdownView ', () => {
    let selectedOption = '';
    const props = {
        name: 'Dropdown-1',
        options: ['option 1', 'option 2'],
        onChangeHandler: value => {
            selectedOption = value;
        },
        isValid: true
    };

    let wrapper = null;

    it('Failed validation display required message', () => {
        wrapper = mount(<DropDownView {...props} />);
        expect(wrapper.find('p').text()).toBe('*required');
        expect(wrapper).toMatchSnapshot();
    });

    it('Success validation does not display required message', () => {
        wrapper = mount(<DropDownView {...props} isValid={false} />);
        expect(wrapper.find('p')).toEqual({});
        expect(wrapper).toMatchSnapshot();
    });

    afterEach(() => {
        wrapper && wrapper.unmount();
    });
});
