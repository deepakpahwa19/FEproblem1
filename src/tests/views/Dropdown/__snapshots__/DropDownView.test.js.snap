// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownView  Failed validation display required message 1`] = `ReactWrapper {}`;

exports[`DropdownView  Success validation does not display required message 1`] = `ReactWrapper {}`;
